### ISSUE TITLE
datetime64("now") is undocumented

### ISSUE THREAD

**Issue by damiendr** [numpy/numpy#10003]
I found out that you can get the current UTC date with `np.datetime64("now")`.

However this isn't documented [here](https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.datetime.html) and there are a few things about the behaviour that aren't obvious. For instance it seems that the precision cannot be greater than one second:

```
In [95]: np.datetime64("now", "ms")
Out[95]: numpy.datetime64('2017-11-11T08:10:31.000')
```

**Comment by shoyer** [comment]
See here for the underlying C-function:
https://github.com/numpy/numpy/blob/65828136fa42188115691475e8736cae089e2113/numpy/core/src/multiarray/datetime_strings.c#L189-L219

Note that it also supports the string "today" indicating a local time.

If we're going to document it, we should probably make sure that the functionality makes sense. In the most recent version of datetime64, we aim for compatibility with Python's datetime module (without timezones). So it's a little strange that "now" means UTC rather than local time (see https://docs.python.org/2/library/datetime.html#datetime.datetime.now) Arguably, we should use something different like "utcnow" for UTC.

The precision issue could potentially be improved, possibly by switching to rely on Python's `time.time()` rather than [our own implementation](https://github.com/numpy/numpy/blob/65828136fa42188115691475e8736cae089e2113/numpy/core/src/multiarray/datetime_strings.c#L75). For now it would indeed make sense to document the (potentially) limited precision.

**Comment by mohdsanadzakirizvi** [comment]
Would like to work on this as my first contribution, is it open ?

**Comment by shoyer** [comment]
As noted above, I think we should probably clean up this functionality before documenting it, which would encourage more people to rely on functionality that would likely change.

**Comment by jorijnsmit** [comment]
Yeah the precision thing is quite misleading I think, passing `'ms'` only increases the size of the float.

**Comment by Shreyjaradi** [comment]
I have raised an PR to add the example of ```np.datetime64("now", "ms")```
 https://github.com/numpy/numpy/pull/26477
